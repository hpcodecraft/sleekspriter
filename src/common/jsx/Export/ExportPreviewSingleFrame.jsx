var ExportPreviewSingleFrame = React.createClass({
  mixins: [PostalSubscriptionMixin, FrameCanvasMixin],
  render: function() {
    var classes = React.addons.classSet({
      preview: true,
      checkerboard: (this.props.format === 'png' || this.props.format === 'gif') ? true : false,
    });
    return (
      <canvas className={classes} width={this.props.frameSize.width} height={this.props.frameSize.height} />
    )
  }
});