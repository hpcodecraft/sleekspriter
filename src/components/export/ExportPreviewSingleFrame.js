import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { FrameCanvas } from "../canvases";

const ExportPreviewSingleFrame = ({ format, frame, size, pixels, zoom }) => {
  const classes = {
      preview: true,
      checkerboard: format === "png" || format === "gif" ? true : false
    },
    style = {
      width: size.width * zoom,
      height: size.height * zoom
    },
    background = format === "jpeg" ? "#ffffff" : null;

  return (
    <div className={classnames(classes)} style={style}>
      <FrameCanvas
        frame={frame}
        size={size}
        zoom={zoom}
        pixels={pixels[frame] || null}
        background={background}
      />
    </div>
  );
};

ExportPreviewSingleFrame.propTypes = {
  format: PropTypes.string.isRequired,
  frame: PropTypes.number.isRequired,
  size: PropTypes.object.isRequired,
  pixels: PropTypes.object.isRequired,
  zoom: PropTypes.number.isRequired
};

export default ExportPreviewSingleFrame;
