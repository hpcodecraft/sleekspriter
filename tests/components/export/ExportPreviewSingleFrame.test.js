import ExportPreviewSingleFrame from "components/export/ExportPreviewSingleFrame";

describe("ExportPreviewSingleFrame", () => {
  let props, jsx, wrapper;
  beforeEach(() => {
    props = {
      format: "png",
      frame: 1,
      layers: [],
      size: { width: 16, height: 16 },
      pixels: {},
      zoom: 1,
    };
    jsx = <ExportPreviewSingleFrame {...props} />;
    wrapper = shallow(jsx);
  });

  it("should render correctly", () => {
    expect(render(jsx)).toMatchSnapshot();
  });

  describe("PNG preview", () => {
    it("does have a checkerboard background", () => {
      expect(wrapper.hasClass("checkerboard")).toEqual(true);
    });
  });

  describe("JPG preview", () => {
    it("does not have a checkerboard background", () => {
      props.format = "jpeg";
      wrapper = shallow(<ExportPreviewSingleFrame {...props} />);
      expect(wrapper.hasClass("checkerboard")).toEqual(false);
    });
  });

  describe("GIF preview", () => {
    it("does have a checkerboard background", () => {
      props.format = "gif";
      wrapper = shallow(<ExportPreviewSingleFrame {...props} />);
      expect(wrapper.hasClass("checkerboard")).toEqual(true);
    });
  });
});
