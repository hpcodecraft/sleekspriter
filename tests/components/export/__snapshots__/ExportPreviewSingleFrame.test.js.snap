// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportPreviewSingleFrame should render correctly 1`] = `"<div class=\\"preview checkerboard\\" style=\\"width:16px;height:16px;\\"><canvas width=\\"16\\" height=\\"16\\" style=\\"width:16px;height:16px;\\"></canvas></div>"`;
